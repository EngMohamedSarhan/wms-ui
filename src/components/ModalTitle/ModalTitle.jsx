import React from "react";
import PropTypes from "prop-types";
import { DialogTitle, IconButton } from "@mui/material";
import { Close } from "@mui/icons-material";

const ModalTitle = ({ title, onClose, sx, ...props }) => (
  <DialogTitle
    {...props}
    sx={{ m: 0, p: 2, minHeight: 50, color: "primary.main", ...sx }}
  >
    {title}
    <IconButton
      aria-label="close"
      onClick={onClose}
      sx={{
        position: "absolute",
        right: 8,
        top: 8,
        color: (theme) => theme.palette.grey[500],
      }}
    >
      <Close />
    </IconButton>
  </DialogTitle>
);

ModalTitle.propTypes = {
  title: PropTypes.string,
  onClose: PropTypes.func.isRequired,
  sx: PropTypes.object,
};

export default ModalTitle;
