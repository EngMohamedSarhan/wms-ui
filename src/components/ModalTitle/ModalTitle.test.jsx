/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import React from "react";
import ShallowRenderer from "react-shallow-renderer";

import ModalTitle from "./ModalTitle";

describe("ModalTitle Component", () =>
  test("ModalTitle rendering", () => {
    const onClose = jest.fn();
    const { queryByText } = render(<ModalTitle title="test" onClose={onClose} />);
    const modalTitle = queryByText("test");
    expect(modalTitle).toBeInTheDocument();
  }));

describe("ModalTitle Component", () =>
  test("ModalTitle snapshot", () => {
    const onClose = jest.fn();
    const renderer = new ShallowRenderer();
    renderer.render(<ModalTitle title="test" onClose={onClose} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  }));
