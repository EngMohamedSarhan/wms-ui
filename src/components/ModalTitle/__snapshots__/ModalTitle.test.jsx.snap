// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot Modal snapshot 1`] = `
<ForwardRef(DialogTitle)
  sx={
    Object {
      "color": "primary.main",
      "m": 0,
      "minHeight": 50,
      "p": 2,
    }
  }
>
  test
  <ForwardRef(IconButton)
    aria-label="close"
    onClick={[MockFunction]}
    sx={
      Object {
        "color": [Function],
        "position": "absolute",
        "right": 8,
        "top": 8,
      }
    }
  >
    <Memo />
  </ForwardRef(IconButton)>
</ForwardRef(DialogTitle)>
`;

exports[`ModalTitle Component ModalTitle snapshot 1`] = `
<ForwardRef(DialogTitle)
  sx={
    Object {
      "color": "primary.main",
      "m": 0,
      "minHeight": 50,
      "p": 2,
    }
  }
>
  test
  <ForwardRef(IconButton)
    aria-label="close"
    onClick={[MockFunction]}
    sx={
      Object {
        "color": [Function],
        "position": "absolute",
        "right": 8,
        "top": 8,
      }
    }
  >
    <Memo />
  </ForwardRef(IconButton)>
</ForwardRef(DialogTitle)>
`;

exports[`ModalTitle snapshot ModalTitle snapshot 1`] = `
<ForwardRef(DialogTitle)
  sx={
    Object {
      "color": "primary.main",
      "m": 0,
      "minHeight": 50,
      "p": 2,
    }
  }
>
  test
  <ForwardRef(IconButton)
    aria-label="close"
    onClick={[MockFunction]}
    sx={
      Object {
        "color": [Function],
        "position": "absolute",
        "right": 8,
        "top": 8,
      }
    }
  >
    <Memo />
  </ForwardRef(IconButton)>
</ForwardRef(DialogTitle)>
`;
